server/tests.py

```diff
@@ -182,54 +188,54 @@ def test_comments(test_client: FlaskClient):
         follow_redirects=True,
     )
     assert r.status_code == 200
-    r = test_client.get("/authors/1/posts/1/comments", follow_redirects=True)
+    r = test_client.get( f"/authors/{T_USER1_UUID}/posts/{first_post_id}/comments", follow_redirects=True)
     data = json.loads(r.data.decode("utf-8"))
     assert data["type"] == "comments"
     assert data["size"] == 1
 
 
 @pytest.mark.order(7)
 def test_followers(test_client: FlaskClient):
-    r = test_client.get("/authors/2/followers", follow_redirects=True)
+    r = test_client.get(f"/authors/{T_USER2_UUID}/followers", follow_redirects=True)
     data = json.loads(r.data.decode("utf-8"))
     assert data["type"] == "followers"
     assert len(data["items"]) == 0
-    r = test_client.put("/authors/2/followers/1", follow_redirects=True)
+    r = test_client.put(f"/authors/{T_USER2_UUID}/followers/{T_USER1_UUID}", follow_redirects=True)
     assert r.status_code == 200
-    r = test_client.put("/authors/2/followers/3", follow_redirects=True)
+    r = test_client.put(f"/authors/{T_USER2_UUID}/followers/{T_USER3_UUID}", follow_redirects=True)
     assert r.status_code == 403
-    r = test_client.get("/authors/2/followers", follow_redirects=True)
+    r = test_client.get(f"/authors/{T_USER2_UUID}/followers", follow_redirects=True)
     data = json.loads(r.data.decode("utf-8"))
     assert data["type"] == "followers"
     assert len(data["items"]) == 1
-    r = test_client.get("/authors/2/followers/1", follow_redirects=True)
+    r = test_client.get(f"/authors/{T_USER2_UUID}/followers/{T_USER1_UUID}", follow_redirects=True)
     assert r.status_code == 200
     data = json.loads(r.data.decode("utf-8"))
     assert len(data["items"]) == 1
-    r = test_client.delete("/authors/2/followers/1", follow_redirects=True)
+    r = test_client.delete(f"/authors/{T_USER2_UUID}/followers/{T_USER1_UUID}", follow_redirects=True)
     assert r.status_code == 204
-    r = test_client.get("/authors/2/followers/1", follow_redirects=True)
+    r = test_client.get(f"/authors/{T_USER2_UUID}/followers/{T_USER1_UUID}", follow_redirects=True)
     assert r.status_code == 200
     data = json.loads(r.data.decode("utf-8"))
     assert len(data["items"]) == 0
 
 
 @pytest.mark.order(8)
 def test_inbox(test_client: FlaskClient):
-    r = test_client.get("/authors/2/inbox", follow_redirects=True)
+    r = test_client.get(f"/authors/{T_USER2_UUID}/inbox", follow_redirects=True)
     assert r.status_code == 401
     # user 1 will follow user 2
-    test_client.put("/authors/2/followers/1", follow_redirects=True)
+    test_client.put(f"/authors/{T_USER2_UUID}/followers/{T_USER1_UUID}", follow_redirects=True)
     logout(test_client)
-    login(test_client, 2)
-    r = test_client.get("/authors/2/inbox", follow_redirects=True)
+    login(test_client, T_USER2_UUID)
+    r = test_client.get(f"/authors/{T_USER2_UUID}/inbox", follow_redirects=True)
```

Comment:
For getting followers there's an issue that was exposed because of the change to UUIDs and not using simple numbers. The tests seem to have been passing because the IDs of authors and follow objects were shared (both 1 in this case).


When adding a follow object reference to the inbox, the id of the author is stored; not the id of the follow object ([api.py line 352](https://github.com/giancarlopernudisegura/cmput404/pull/120/files#diff-a99f7f8bd872c86c77b97baf160a74f59f24473061dcab699ba028f1a1685653R352)).

If you look under [models.py line 366](https://github.com/giancarlopernudisegura/cmput404/pull/120/files#diff-d97741605a183dfd2c9975818dd030d80edfef76b2dd3fa5f696604068146b8aR366) it looks for a request object and not an author. This is causing the tests to fail since the id is an author reference and not a request reference.  Thus inbox attempts to call .json() on a None object.

I'd like some help from gino on this since he is the one who wrote both the test and the follow/inbox endpoints.

The error in question:
``` 
E           AttributeError: 'NoneType' object has no attribute 'json'
models.py:367: AttributeError```

Reply:
Seems like the line in 352 is a typo then. We should always be using the ID reference to the follower object, not the author. Good catch.